e2e/backend_service_update_test.go

```diff
@@ -0,0 +1,200 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/exec"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/backendservice"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/tcproute"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/workflow/plan"
+	"github.com/kr/pretty"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/networkservices/v1"
+)
+
+func buildTcpRoute(builder *rgraph.Builder, name string, serviceName string) (*cloud.ResourceID, error) {
```

Comment:
@kl52752 Is the plan here to only test `INTERNAL_SELF_MANAGED` BackendServices?

TCPRoute test already creates this type of BackendService. Testing a different LB scheme in the BackendService should be more helpful.

Reply:
I think those tests should be "neutral" to GCE LoadBalancers and check different capabilities of rGraph. 
For example in this test we check how Backend Service recreation impacts parents and it doesn't and to check this we need to either change LoadBalancing Schema either Network. It doesn't really matter what is the value of LoadBalancingSchema.

Reply:
+1, it shouldn't matter